validation/schema.py

```diff
@@ -0,0 +1,63 @@
+import json
+import os
+from typing import List
+
+import requests
+
+
+class SchemaValidation:
+    schema_by_type = {}
+
+    def __init__(self, validator_url):
+        self.validator_url = validator_url
+        self.__load_schema_files()
+
+    def validate_data(self, data):
+        issues = {}
+        for entities in data:
+            for entity_type, entity in entities.items():
+                if entity_type == 'row':
+                    continue
+                human_errors = self.get_human_errors(entity_type, entity)
+                if human_errors:
+                    entity.setdefault('errors', []).extend(human_errors)
+                    issues.setdefault(str(entities['row']), []).extend(human_errors)
+        return issues
+
+    def get_human_errors(self, entity_type, entity):
+        validation_result = self.validate(entity_type, entity)
+        return self.__translate_to_human(entity_type, validation_result)
+
+    def validate(self, entity_type, entity):
+        schema = self.schema_by_type.get(entity_type, {})
+        return self.__validate(schema, entity)
+
+    def __validate(self, schema, entity):
+        schema.pop('id', None)
+        payload = self.__create_validator_payload(schema, entity)
+        return requests.post(self.validator_url, json=payload).json()
+
+    def __load_schema_files(self):
+        types = ['study', 'sample', 'run_experiment', 'isolate_genome_assembly_information']
```

Comment:
This list of entity types coupled to this method. I would move it out to a config file.
also right now we only have 1 schema for a sample or any of these types, but soon we have more possible schema for at least sample. Based on this code it won't work.